pkg/services/network/netop_provider.go

```diff
@@ -122,16 +124,25 @@ func (np *netopNetworkProvider) ConfigureVirtualMachine(ctx context.Context, clu
 		return err
 	}
 
-	for _, vnif := range vm.Spec.NetworkInterfaces {
-		if vnif.NetworkType == string(network.Spec.Type) && vnif.NetworkName == network.Name {
+	if vm.Spec.Network == nil {
+		vm.Spec.Network = &vmoprv1.VirtualMachineNetworkSpec{}
+	}
+	for _, vnif := range vm.Spec.Network.Interfaces {
+		if vnif.Network.TypeMeta.GroupVersionKind() == netopv1.SchemeGroupVersion.WithKind("Network") && vnif.Network.Name == network.Name {
```

Comment:
Can we define a const for the network type?

Reply:
The const is more or less the api struct, but it cannot be used.

Do we need this somewhere else or why do you think we should consider a struct here?

Reply:
Note: I've got no strong opinion on it, I am okay to have a const, just want to clarify.

Reply:
Each network provider has its own network type: Network, VirtualNetwork and SubnetSet. Maybe it is easy to understand the mapping if we have a network type const or have an method GetNetworkType() in the NetworkProvider interface.

Reply:
added gvk's per type to use instead 👍